ext/DecapodesCUDAExt.jl

```diff
@@ -0,0 +1,125 @@
+module DecapodesCUDAExt
+using CombinatorialSpaces
+using LinearAlgebra
+using Base.Iterators
+using Catlab
+using Krylov
+using CUDA
+using CUDA.CUSPARSE
+using MLStyle
+import Decapodes: default_dec_cu_matrix_generate
+
+function default_dec_cu_matrix_generate(sd, my_symbol, hodge)
+  op = @match my_symbol begin
+
+    # Regular Hodge Stars
+    :⋆₀ => dec_cu_mat_hodge(0, sd, hodge)
+    :⋆₁ => dec_cu_mat_hodge(Val{1}, sd, hodge)
+    :⋆₂ => dec_cu_mat_hodge(2, sd, hodge)
+
+    # Inverse Hodge Stars
+    :⋆₀⁻¹ => dec_cu_mat_inverse_hodge(0, sd, hodge)
+    :⋆₁⁻¹ => dec_cu_pair_inv_hodge(Val{1}, sd, hodge) # Special since Geo is a solver
+    :⋆₂⁻¹ => dec_cu_mat_inverse_hodge(1, sd, hodge)
+
+    # Differentials
+    :d₀ => dec_cu_mat_differential(0, sd)
+    :d₁ => dec_cu_mat_differential(1, sd)
+
+    # Dual Differentials
+    :dual_d₀ || :d̃₀ => dec_cu_mat_dual_differential(0, sd)
+    :dual_d₁ || :d̃₁ => dec_cu_mat_dual_differential(1, sd)
+
+    # Wedge Products
+    :∧₀₁ => dec_cu_pair_wedge_product(Tuple{0,1}, sd)
+    :∧₁₀ => dec_cu_pair_wedge_product(Tuple{1,0}, sd)
+    :∧₀₂ => dec_cu_pair_wedge_product(Tuple{0,2}, sd)
+    :∧₂₀ => dec_cu_pair_wedge_product(Tuple{2,0}, sd)
+    :∧₁₁ => dec_cu_pair_wedge_product(Tuple{1,1}, sd)
+
+    _ => error("Unmatched operator $my_symbol")
+  end
+
+  return op
+end
+
+# TODO: Update this to better cast hodges
+function dec_cu_mat_hodge(k, sd::HasDeltaSet, hodge)
```

Comment:
I think the issue here was that I should be multi-dispatching on ```dec_mat_hodge```, between CPU and CUDA version, but I think this is moot since this version seems preferable to me.